Kitodo/src/main/java/org/kitodo/production/services/index/IndexingService.java

```diff
@@ -0,0 +1,567 @@
+     * state' if any object type is in 'no state' state - return 'indexing
+     * successful' state if all object types are in 'indexing successful' state
+     *
+     * @return static variable for global indexing state
+     */
+    public IndexStates getAllObjectsIndexingState() {
+        for (ObjectType objectType : objectTypes) {
+            if (Objects.equals(objectIndexingStates.get(objectType), IndexStates.INDEXING_FAILED)) {
+                return IndexStates.INDEXING_FAILED;
+            }
+            if (Objects.equals(objectIndexingStates.get(objectType), IndexStates.NO_STATE)) {
+                return IndexStates.NO_STATE;
+            }
+        }
+        return IndexStates.INDEXING_SUCCESSFUL;
+    }
+
+    /**
+     * Return whether any indexing process is currently in progress or not.
+     *
+     * @return boolean Value indicating whether any indexing process is currently in
+     *         progress or not
+     */
+    public boolean indexingInProgress() {
+        return !Objects.equals(this.currentIndexState, ObjectType.NONE) || indexingAll;
+    }
+
+    /**
+     * Check if current mapping is empty.
+     *
+     * @return true if mapping is empty, otherwise false
+     */
+    public boolean isMappingEmpty() {
+        String emptyMapping = "{\n\"mappings\": {\n\n    }\n}";
+        return isMappingEqualTo(emptyMapping);
+    }
+
+    private boolean isMappingEqualTo(String mapping) {
+        try (JsonReader mappingExpectedReader = Json.createReader(new StringReader(mapping));
+             JsonReader mappingCurrentReader = Json.createReader(new StringReader(indexRestClient.getMapping()))) {
+            JsonObject mappingExpected = mappingExpectedReader.readObject();
+            JsonObject mappingCurrent = mappingCurrentReader.readObject().getJsonObject(indexRestClient.getIndex());
+            return mappingExpected.equals(mappingCurrent);
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    private static String readMapping() {
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        try (InputStream inputStream = classloader.getResourceAsStream("mapping.json")) {
+            if (Objects.nonNull(inputStream)) {
+                String mapping = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+                try (JsonReader jsonReader = Json.createReader(new StringReader(mapping))) {
+                    JsonObject jsonObject = jsonReader.readObject();
+                    return jsonObject.toString();
+                }
+            } else {
+                Helper.setErrorMessage("Mapping not found!");
+                return "";
+            }
+        } catch (IOException e) {
+            Helper.setErrorMessage(e.getLocalizedMessage(), logger, e);
+            return "";
+        }
+    }
+
+    private static IndexRestClient indexRestClient = IndexRestClient.getInstance();
```

Comment:
Can you move this on top of the class?

Reply:
Done.

Reply:
That line is still right here in the middle of the class and not at the top where declarations should be.

Reply:
Done (for real, this time!)